20_hanoi/README.md

```diff
@@ -0,0 +1,39 @@
+# Exercise 20 - hanoi
+
+Good job, you have come a very long way. You should be proud of yourself for making it this far. At this point, you hopefully feel comfortable
+with recursion, and so, this **final recursive exercise** should give you a proper challenge and take your abilities to the _next_ level.
+
+Tower of Hanoi is a classic challenge in which we have 3 towers. Tower 2 and tower 3 are empty.
+Tower 1 has `n` disks. The goal is to move all disks from tower 1 to tower 3 so that they are exactly the in the same order. For instance:
+
+```javascript
+[[3, 2, 1], [], []] 
+
+// ...
+
+[[], [], [3, 2, 1]]
+```
+
+The rules are as follows:
+- Each disk has a length. For abstraction purposes, we will imagine each disk as an integer with that disks' length.
+- We can only move 1 disk at a time between any of the towers
+- The towers are modelled as stacks. We can move disks from the top and add them to the top with the `Array.prototype.pop` and `Array.prototype.push` methods.
+- **Disks can only be placed on top of disks that are smaller**. For instance, we cannot have the following tower: `[3, 4]`. It would violate the rules
+
+Your task is to create a function, `hanoi`, that when given an integer `n` will perform the required operations to transport all the numbers from the `first tower`
+onto the `third tower`.
```

Comment:
I think this bit could be made clearer as to *exactly* what's expected from the learner from their `hanoi` function.

e.g. (wording suggestion only)
> Your task is to create a function, `hanoi`, that when given the number of disks in the starting tower, will return an array containing the state of the towers after every "move". The first element will be the towers' initial state and the last element will be the towers' final state.

**BIG** consideration (will expand in a comment on the tests file) - whether we want to enforce solving in the minimum number of moves, or if any solution that goes from start to finish shall be accepted.